cmd/promtail/main.go

```diff
@@ -37,6 +37,12 @@ func main() {
 		}
 	}
 
+	err := config.ClientConfig.Client.Validate()
+	if err != nil {
+		level.Error(util.Logger).Log("msg", "error validating client config", "error", err)
+		os.Exit(1)
+	}
```

Comment:
Could you move this code into promtail/client/client.go:79 (New) please?  That way other users will benefit from it.